From fc1d26dd08c48c04fc9883b36a94c219aba2091c Mon Sep 17 00:00:00 2001
From: Luca Boccassi <luca.boccassi@microsoft.com>
Date: Mon, 28 Oct 2019 14:31:38 +0000
Subject: [PATCH] launcher: fix build with musl libc
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

../src/launch/launcher.c: In function ‘launcher_fork’:
../src/launch/launcher.c:378:60: error: ‘WEXITED’ undeclared (first use in this function); did you mean ‘WIFEXITED’?
         r = sd_event_add_child(launcher->event, NULL, pid, WEXITED, launcher_on_child_exit, launcher);
                                                            ^~~~~~~
                                                            WIFEXITED

Include sys/wait.h which defines it.

Signed-off-by: Luca Boccassi <luca.boccassi@microsoft.com>
Suggested-by: Khem Raj <raj.khem@gmail.com>
Upstream-Status: pending https://github.com/bus1/dbus-broker/pull/214
---
 src/launch/launcher.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/launch/launcher.c b/src/launch/launcher.c
index 2ec4bda..1f38fcf 100644
--- a/src/launch/launcher.c
+++ b/src/launch/launcher.c
@@ -10,6 +10,7 @@
 #include <stdlib.h>
 #include <sys/prctl.h>
 #include <sys/un.h>
+#include <sys/wait.h>
 #include <systemd/sd-bus.h>
 #include <systemd/sd-daemon.h>
 #include <systemd/sd-event.h>
-- 
2.20.1

